meta_yaml_schema/schema.py

```diff
@@ -0,0 +1,96 @@
+schema = {
+    "type": "object",
+    "required": [
+        "title",
+        "description",
+        "pangeo_forge_version",
+        "pangeo_notebook_version",
+        "recipes",
+        "provenance",
+        "maintainers",
+        "bakery",
+    ],
+    "properties": {
+        "title": {"type": "string"},
+        "description": {"type": "string"},
+        # TODO: semantic version format
+        "pangeo_forge_version": {"type": "string"},
+        # TODO: convert to image tag
+        "pangeo_notebook_version": {"type": "string"},
```

Comment:
@cisaacstern, i'm curious... Would it be possible to define this via Pydantic and then export the json schema from the Pydantic model instead of manually generating it? 

```python
In [1]: import pydantic

In [2]: class Maintainer(pydantic.BaseModel):
   ...:     name: str
   ...:     orcid: str
   ...:     github: str
   ...: 

In [3]: class PangeoForge(pydantic.BaseModel):
   ...:     title: str
   ...:     description: str
   ...:     maintainers: list[Maintainer]
   ...: 
```

```python
In [10]: PangeoForge.schema()
Out[10]: 
{'title': 'PangeoForge',
 'type': 'object',
 'properties': {'title': {'title': 'Title', 'type': 'string'},
  'description': {'title': 'Description', 'type': 'string'},
  'maintainers': {'title': 'Maintainers',
   'type': 'array',
   'items': {'$ref': '#/definitions/Maintainer'}}},
 'required': ['title', 'description', 'maintainers'],
 'definitions': {'Maintainer': {'title': 'Maintainer',
   'type': 'object',
   'properties': {'name': {'title': 'Name', 'type': 'string'},
    'orcid': {'title': 'Orcid', 'type': 'string'},
    'github': {'title': 'Github', 'type': 'string'}},
   'required': ['name', 'orcid', 'github']}}}

In [11]: PangeoForge.schema_json()
Out[11]: '{"title": "PangeoForge", "type": "object", "properties": {"title": {"title": "Title", "type": "string"}, "description": {"title": "Description", "type": "string"}, "maintainers": {"title": "Maintainers", "type": "array", "items": {"$ref": "#/definitions/Maintainer"}}}, "required": ["title", "description", "maintainers"], "definitions": {"Maintainer": {"title": "Maintainer", "type": "object", "properties": {"name": {"title": "Name", "type": "string"}, "orcid": {"title": "Orcid", "type": "string"}, "github": {"title": "Github", "type": "string"}}, "required": ["name", "orcid", "github"]}}}'

```

Reply:
for documentation, we could use https://pypi.org/project/autodoc-pydantic/ which integrates well with sphinx...

Reply:
> Considering what (if any) integration we want with pangeo-forge-runner (or perhaps we will leave that layer naive to this schema, and let orchestrator do the validation).

w/ a Pydantic model, the validation would also be streamlined since Pydantic would take care of this wherever the model is used.